receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -217,8 +227,11 @@ func (mf *metricFamily) ToMetric() *metricspb.Metric {
 			},
 			Timeseries: timeseries,
 		}
+		if mf.nodeResource != nil {
```

Comment:
Nit: this check is not needed. It should be simpler to just do:
return &metricspb.Metric{... Resource: mf.nodeResource}

Reply:
Done.